examples/device-dashboard/web_root/components.js

```diff
@@ -200,3 +215,61 @@ export function Setting(props) {
   <//>
 <//>`;
 };
+
+export function Toast({ok, text, close}) {
+  const closebtn = useRef(null);
+  const from = 'translate-y-2 opacity-0 sm:translate-y-0 sm:translate-x-2';
+  const to = 'translate-y-0 opacity-100 sm:translate-x-0';
+  const [tr, setTr] = useState(from);
+  useEffect(function() {
+    setTr(to); 
+    setTimeout(ev => closebtn && closebtn.current.click && closebtn.current.click(), 1500);
+  }, []);
+  const onclose = ev => { setTr(from); setTimeout(close, 300); };
+  return html`
+<div aria-live="assertive" class="z-10 pointer-events-none absolute inset-0 flex items-end px-4 py-6 sm:items-start sm:p-6">
+  <div class="flex w-full flex-col items-center space-y-4 sm:items-end">
+    <div class="pointer-events-auto w-full max-w-sm overflow-hidden rounded-lg bg-white shadow-lg ring-1 ring-black ring-opacity-5 transform ease-out duration-300 transition ${tr}">
+      <div class="p-4">
+        <div class="flex items-start">
+          <div class="flex-shrink-0">
+            <${ok ? Icons.ok : Icons.fail} class="h-6 w-6 ${ok ? 'text-green-400' : 'text-red-400'}" />
+          <//>
+          <div class="ml-3 w-0 flex-1 pt-0.5">
+            <p class="text-sm font-medium text-gray-900">${text}</p>
+            <p class="hidden mt-1 text-sm text-gray-500">Anyone with a link can now view this file.</p>
+          <//>
+          <div class="ml-4 flex flex-shrink-0">
+            <button type="button" ref=${closebtn} onclick=${onclose} class="inline-flex rounded-md bg-white text-gray-400 hover:text-gray-500 focus:outline-none">
+              <span class="sr-only">Close</span>
+              <svg class="h-5 w-5" viewBox="0 0 20 20" fill="currentColor" aria-hidden="true">
+                <path d="M6.28 5.22a.75.75 0 00-1.06 1.06L8.94 10l-3.72 3.72a.75.75 0 101.06 1.06L10 11.06l3.72 3.72a.75.75 0 101.06-1.06L11.06 10l3.72-3.72a.75.75 0 00-1.06-1.06L10 8.94 6.28 5.22z" />
+              <//>
+            <//>
+          <//>
+        <//>
+      <//>
+    <//>
+  <//>
+<//>`;
+};
+
+export function Header({logout, user, setShowSidebar, showSidebar}) {
```

Comment:
Likewise, the main layout of the dashboard - header, sidebar, should be in main.js, because high chances are , they will be customized.
Please leave it in main.js